core/src/main/java/com/loop54/model/request/GetPopularEntitiesRequest.java

```diff
@@ -2,52 +2,41 @@
 
 import java.util.List;
 
-import com.loop54.model.request.parameters.EntityCollectionParameters;
-
 /** This class used to perform a request to get entities that a user (or all users) most commonly interacted with or navigated to. */
-public class GetPopularEntitiesRequest extends Request {
+public class GetPopularEntitiesRequest extends GetPopularOrRecentEntitiesRequest {
 
     /**
      * @param  behaviorType interaction or navigation type to include (such as {@literal click}, {@literal purchase} or {@literal search}).
-     * @param  forUserId    user ID (normally the same as the one in the User-Id header) to retrieve the most common entities for that user or null to retrieve the globally most common entities.
+     * @param  forUserId    a user ID to retrieve the most common/recent entities for that user or null to retrieve the globally most common/recent entities.
+     * {@link GetGetPopularOrRecentEntitiesRequest#CURRENT_USER_PLACEHOLDER} can be specified to use the user ID from the User-Id request header.
      * @param  entityType   entity types to include (such as {@literal Product} or {@literal Query})
      */
     public GetPopularEntitiesRequest(String behaviorType, String forUserId, List<String> entityType) {
-        this.behaviorType = behaviorType;
-        this.forUserId = forUserId;
-        this.entityType = entityType;
+        super(behaviorType, forUserId, entityType);
     }
 
     /**
      * Retrieve the globally most common products for a specfic user
```

Comment:
Existing comment, I know, but "products" is a bit misleading here, since the entity type could be anything. It could be the most popular categories, for example. So you could just write "entities" or maybe "entities (such as products)" if you want to be more helpful.